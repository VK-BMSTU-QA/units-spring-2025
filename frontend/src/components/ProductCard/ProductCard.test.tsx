import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ProductCard } from './ProductCard';

afterEach(jest.clearAllMocks);

describe('Product card test', () => {
    it('should render correctly', () => {
        const rendered = render(
            <ProductCard
                name="Test Card"
                description="Card for testing"
                price={1234}
                priceSymbol="$"
                category="Для дома"
                id={1234}
            />
        );

        expect(rendered.asFragment()).toMatchSnapshot();
    });

    it('should not contain image when no imgUrl', () => {
        const rendered = render(
            <ProductCard
                name="Test Card"
                description="Card for testing"
                price={1234}
                priceSymbol="$"
                category="Для дома"
                id={1234}
            />
        );

        expect(rendered.baseElement).not.toContainHTML('img');
    });

    it('should contain image when imgUrl is provided', () => {
        const rendered = render(
            <ProductCard
                name="Test Card"
                description="Card for testing"
                price={1234}
                priceSymbol="$"
                category="Для дома"
                id={1234}
                imgUrl="/iphone.png"
            />
        );
        expect(rendered.baseElement).toContainHTML('img');
    });
});
