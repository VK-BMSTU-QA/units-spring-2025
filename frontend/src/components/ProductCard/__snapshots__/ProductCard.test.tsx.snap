// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard test should render correctly 1`] = `
<DocumentFragment>
  <div
    class="product-card"
  >
    <div
      class="product-card__text"
    >
      <h2
        class="product-card__name"
      >
        product
      </h2>
      <div
        class="product-card__description"
      >
        product description
      </div>
      <h1
        class="product-card__price"
      />
      <div
        class="product-card__category"
      >
        Одежда
      </div>
    </div>
    <img
      alt="product"
      class="product-card__image"
      src="/lamp.png"
    />
  </div>
</DocumentFragment>
`;
